frontend/src/metabase/lib/compose-event-handlers.ts

```diff
@@ -0,0 +1,9 @@
+type HandlerType<E> = (event: E) => void;
+
+export const composeEventHandlers = <E>(...handlers: HandlerType<E>[]) => {
+  return function handleEvent(event: E) {
+    handlers.forEach(handler => {
+      handler?.(event);
```

Comment:
Are you sure this `fn?(e)` notation works as expected? I used that at `NativeQueryEditor`, but it was still trying to call the optional callbacks (see [the PR](https://github.com/metabase/metabase/pull/20160/files)). But maybe I just messed up the syntax

Reply:
In your case instead of `this.props?.onClose();` you needed to write `this.props.onClose?.();` — the syntax is a bit unusual though. Actually, it is a good point the check with `typeof` because in some places code passes false instead `undefined` to callback props, thanks for the suggestion.

Reply:
Cool, glad it was helpful. And thanks for the explanation, I knew I somehow messed the syntax